GenericNode/src/main/java/genericnode/servers/Store.java

```diff
@@ -0,0 +1,19 @@
+package genericnode.servers;
+
+import java.rmi.Remote;
+import java.rmi.RemoteException;
+import java.util.ArrayList;
+
+// A remote interface for RMI server
+public interface Store extends Remote {
+
+    void put(String key, String value) throws RemoteException;
+
+    String get(String key) throws RemoteException;
+
+    void del(String key) throws RemoteException;
+
+    ArrayList<String> store() throws RemoteException;
+
+    void exit() throws RemoteException;
+}
```

Comment:
I believe we could have used the Server.java instead of this new class, Since the only difference between them is the thrown exceptions, you dont need to add these exceptions to function declaration, you can catch them in the final RMIServer class itself instead of throwing them. Also extend RmiServer Class to Remote.